embedded/pwm_driver/pwm_driver.cc

```diff
@@ -37,8 +37,9 @@ Adafruit PCA9685 library: https://github.com/adafruit/Adafruit-PWM-Servo-Driver-
 #define ALLLED_OFF_H 0xFD
 
 
-PwmDriver::PwmDriver(char* dev) {
-	device_name = dev;
+PwmDriver::PwmDriver(const std::string &dev) {
+	device_name = dev.c_str();
```

Comment:
This may not do what you want; since this will get a pointer to a stack object that will be deallocated; I believe this is a dangling reference

You want to do one of the following:
- At the point where device_name_ is normally used, get the c_str
- Allocate a copy and hold onto that in the class

Reply:
did neither, combined `init`